fxa-client/sdks/android/library/src/main/java/org/mozilla/fxaclient/internal/FirefoxAccount.kt

```diff
@@ -0,0 +1,160 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fxaclient.internal
+
+/**
+ * FirefoxAccount represents the authentication state of a client.
+ */
+class FirefoxAccount : RustObject<RawFxAccount> {
+
+    internal constructor(rawPointer: RawFxAccount): super(rawPointer)
+
+    /**
+     * Create a FirefoxAccount using the given config, client id, and redirect uri.
+     *
+     * If the config is passed into this method, calling `close` on it is no longer required
+     * (but may be done if desired).
+     *
+     * This does not make network requests, and can be used on the main thread.
+     */
+    constructor(config: Config, clientId: String, redirectUri: String)
+    : this(config.rustCall { e ->
```

Comment:
Interesting, but guess it makes sense

Reply:
Yep, we're taking ownership of the config object and so need to own it.